Test `UpdateMavenSliceTest.throwsExceptionWhenMetadataUpdatesDoneSimultaneously` is flaky

Test `UpdateMavenSliceTest.throwsExceptionWhenMetadataUpdatesDoneSimultaneously` seems to be flaky.
Usually the test pass, but sometimes it crashes with the following error:

```
[1;31mERRORm] 1;31m  UpdateMavenSliceTest.throwsExceptionWhenMetadataUpdatesDoneSimultaneously:262 All failure due to concurrent lock access
Expected: every item is (an instance of java.lang.IllegalStateException and an exception with message a string containing "Failed to acquire lock.")
     but: an item an instance of java.lang.IllegalStateException <com.artipie.asto.ValueNotFoundException: No value for key: .upload/org/example/artifact/0.1/maven-metadata.xml> is a com.artipie.asto.ValueNotFoundExceptionm
```

I have managed to extract full exception:
```
java.util.concurrent.CompletionException: com.artipie.asto.ValueNotFoundException: No value for key: .upload/org/example/artifact/0.1/maven-metadata.xml
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
	at java.base/java.util.concurrent.CompletableFuture.uniApplyNow(CompletableFuture.java:670)
	at java.base/java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:658)
	at java.base/java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:2165)
	at java.base/java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:143)
	at com.artipie.maven.http.InMemoryStorage.save(InMemoryStorage.java:99)
	at com.artipie.asto.rx.RxStorageWrapper.lambda$save$2(RxStorageWrapper.java:70)
	at io.reactivex.internal.operators.completable.CompletableDefer.subscribeActual(CompletableDefer.java:36)
	at io.reactivex.Completable.subscribe(Completable.java:2309)
	at io.reactivex.internal.operators.completable.CompletableDefer.subscribeActual(CompletableDefer.java:43)
	at io.reactivex.Completable.subscribe(Completable.java:2309)
	at io.reactivex.internal.operators.completable.CompletableMergeIterable.subscribeActual(CompletableMergeIterable.java:91)
	at io.reactivex.Completable.subscribe(Completable.java:2309)
	at io.reactivex.internal.operators.completable.CompletableConcat$CompletableConcatSubscriber.drain(CompletableConcat.java:193)
	at io.reactivex.internal.operators.completable.CompletableConcat$CompletableConcatSubscriber.onNext(CompletableConcat.java:128)
	at io.reactivex.internal.operators.completable.CompletableConcat$CompletableConcatSubscriber.onNext(CompletableConcat.java:43)
	at io.reactivex.internal.operators.flowable.FlowableMap$MapSubscriber.onNext(FlowableMap.java:68)
	at io.reactivex.internal.operators.flowable.FlowableBuffer$PublisherBufferExactSubscriber.onComplete(FlowableBuffer.java:152)
	at io.reactivex.internal.subscribers.BasicFuseableSubscriber.onComplete(BasicFuseableSubscriber.java:120)
	at io.reactivex.internal.operators.flowable.FlowableFromIterable$IteratorSubscription.slowPath(FlowableFromIterable.java:255)
	at io.reactivex.internal.operators.flowable.FlowableFromIterable$BaseRangeSubscription.request(FlowableFromIterable.java:124)
	at io.reactivex.internal.subscribers.BasicFuseableSubscriber.request(BasicFuseableSubscriber.java:153)
	at io.reactivex.internal.operators.flowable.FlowableBuffer$PublisherBufferExactSubscriber.request(FlowableBuffer.java:81)
	at io.reactivex.internal.subscribers.BasicFuseableSubscriber.request(BasicFuseableSubscriber.java:153)
	at io.reactivex.internal.operators.completable.CompletableConcat$CompletableConcatSubscriber.onSubscribe(CompletableConcat.java:116)
	at io.reactivex.internal.subscribers.BasicFuseableSubscriber.onSubscribe(BasicFuseableSubscriber.java:67)
	at io.reactivex.internal.operators.flowable.FlowableBuffer$PublisherBufferExactSubscriber.onSubscribe(FlowableBuffer.java:95)
	at io.reactivex.internal.subscribers.BasicFuseableSubscriber.onSubscribe(BasicFuseableSubscriber.java:67)
	at io.reactivex.internal.operators.flowable.FlowableFromIterable.subscribe(FlowableFromIterable.java:69)
	at io.reactivex.internal.operators.flowable.FlowableFromIterable.subscribeActual(FlowableFromIterable.java:47)
	at io.reactivex.Flowable.subscribe(Flowable.java:14918)
	at io.reactivex.internal.operators.flowable.FlowableMap.subscribeActual(FlowableMap.java:37)
	at io.reactivex.Flowable.subscribe(Flowable.java:14918)
	at io.reactivex.internal.operators.flowable.FlowableBuffer.subscribeActual(FlowableBuffer.java:47)
	at io.reactivex.Flowable.subscribe(Flowable.java:14918)
	at io.reactivex.internal.operators.flowable.FlowableMap.subscribeActual(FlowableMap.java:37)
	at io.reactivex.Flowable.subscribe(Flowable.java:14918)
	at io.reactivex.Flowable.subscribe(Flowable.java:14865)
	at io.reactivex.internal.operators.completable.CompletableConcat.subscribeActual(CompletableConcat.java:40)
	at io.reactivex.Completable.subscribe(Completable.java:2309)
	at io.reactivex.Completable.subscribe(Completable.java:2383)
	at hu.akarnokd.rxjava2.interop.CompletableInterop.lambda$await$1(CompletableInterop.java:47)
	at io.reactivex.Completable.to(Completable.java:2597)
	at com.artipie.asto.Copy.copy(Copy.java:66)
	at com.artipie.maven.http.UpdateMavenSlice.lambda$response$3(UpdateMavenSlice.java:143)
	at com.artipie.asto.UnderLockOperation.lambda$perform$2(UnderLockOperation.java:76)
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1146)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1776)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1763)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
Caused by: com.artipie.asto.ValueNotFoundException: No value for key: .upload/org/example/artifact/0.1/maven-metadata.xml
	at com.artipie.maven.http.InMemoryStorage.value(InMemoryStorage.java:149)
	at com.artipie.asto.SubStorage.value(SubStorage.java:102)
	at com.artipie.asto.rx.RxStorageWrapper.lambda$value$5(RxStorageWrapper.java:88)
	at io.reactivex.internal.operators.single.SingleDefer.subscribeActual(SingleDefer.java:36)
	at io.reactivex.Single.subscribe(Single.java:3603)
	at io.reactivex.internal.operators.single.SingleFlatMapPublisher.subscribeActual(SingleFlatMapPublisher.java:66)
	at io.reactivex.Flowable.subscribe(Flowable.java:14918)
	at io.reactivex.Flowable.subscribe(Flowable.java:14865)
	at com.artipie.asto.Content$From.subscribe(Content.java:131)
	at com.artipie.asto.OneTimePublisher.subscribe(OneTimePublisher.java:62)
	at io.reactivex.internal.operators.flowable.FlowableFromPublisher.subscribeActual(FlowableFromPublisher.java:29)
	at io.reactivex.Flowable.subscribe(Flowable.java:14918)
	at io.reactivex.Flowable.subscribe(Flowable.java:14865)
	at io.reactivex.internal.operators.flowable.FlowableReduceSeedSingle.subscribeActual(FlowableReduceSeedSingle.java:49)
	at io.reactivex.Single.subscribe(Single.java:3603)
	at io.reactivex.Single.subscribe(Single.java:3589)
	at hu.akarnokd.rxjava2.interop.SingleInterop.lambda$get$0(SingleInterop.java:49)
	at io.reactivex.Single.to(Single.java:3872)
	at com.artipie.maven.http.InMemoryStorage.save(InMemoryStorage.java:98)
	... 49 more
```

--- a/src/main/java/com/artipie/maven/http/UpdateMavenSlice.java
+++ b/src/main/java/com/artipie/maven/http/UpdateMavenSlice.java
@@ -132,14 +132,15 @@ final class UpdateMavenSlice implements Slice {
                             valid -> {
                                 final CompletionStage<Response> upd;
                                 if (valid) {
-                                    upd = this.maven.update(
-                                        new Key.From(UpdateMavenSlice.TEMP, location), location
-                                    )
-                                        .thenCompose(nothing -> temp.list(location))
-                                        .thenCompose(
-                                            list -> this.storage.exclusively(
-                                                location,
-                                                target -> new Copy(temp, new ListOf<>(list))
+                                    upd = this.storage.exclusively(
+                                        location,
+                                        target -> this.maven.update(
+                                            new Key.From(UpdateMavenSlice.TEMP, location),
+                                            location
+                                        )
+                                            .thenCompose(nothing -> temp.list(location))
+                                            .thenCompose(
+                                                list -> new Copy(temp, new ListOf<>(list))
                                                     .copy(target).thenCompose(
                                                         nothing ->
                                                             UpdateMavenSlice.remove(temp, list)

